### PR TITLE
feature/h5p-reports-endpoint

### PR DESCRIPTION
**Pull request by rider-griffin**
- creates base ajax endpoint for extraction of report data from wordpress
- the GET request takes an array of content ID's and an array of journey user ID's as a parameter and returns a JSON representation of user records for each content ID supplied

### CODE REVIEW

File: admin/class-h5p-plugin-admin.php [hunk 1]
```diff
@@ -162,6 +165,70 @@ function add_settings_link($links) {
...
+  *
+  */
+  public function ajax_get_users_h5p_report_data() {
+    global $wpdb;
+    $plugin = H5P_Plugin::get_instance();
+    $core = $plugin->get_h5p_instance('core');
+    $action = $_GET['action'];
+    $user_data = array();
+
+    if(!isset($action)) {
```
Comment (zachrip) [inline admin/class-h5p-plugin-admin.php:179]: Can the if statement below handle both cases?

File: admin/class-h5p-plugin-admin.php [hunk 2]
```diff
@@ -162,6 +165,70 @@ function add_settings_link($links) {
...
+  /**
+  * Retrieve a JSON representation of H5P save data for multiple users and contents, via GET request
+  *
+  */
+  public function ajax_get_users_h5p_report_data() {
+    global $wpdb;
+    $plugin = H5P_Plugin::get_instance();
+    $core = $plugin->get_h5p_instance('core');
+    $action = $_GET['action'];
+    $user_data = array();
```
Comment (zachrip) [inline admin/class-h5p-plugin-admin.php:177]: I don't think this var is necessary

File: admin/class-h5p-plugin-admin.php [hunk 3]
```diff
@@ -162,6 +165,70 @@ function add_settings_link($links) {
...
+      return;
+    }
+
+    $json = file_get_contents('php://input');
+    $decoded = json_decode($json);
+    $response = array();
+  
+    try
+    {
+      // The content_ids[] array will be limited in Athena to avoid a timeout
```
Comment (zachrip) [inline admin/class-h5p-plugin-admin.php:195]: This comment isn't really necessary, would make more sense on the other side (in athena)

File: admin/class-h5p-plugin-admin.php [hunk 4]
```diff
@@ -162,6 +165,70 @@ function add_settings_link($links) {
...
+      foreach ($decoded->content_ids as $content_id) {
+        $response["h5p_" . $content_id] = array();
+        
+        foreach ($decoded->journey_user_ids as $journey_user_id) {
+          // Get WordPress user ID corresponding to journey user ID
+          $wp_user_id = $wpdb->get_var($wpdb->prepare(
+            "SELECT id FROM wp_users 
+             WHERE user_login LIKE CONCAT('tcuser', %d)",
```
Comment (zachrip) [inline admin/class-h5p-plugin-admin.php:203]: is there a reason to use LIKE here?

File: admin/class-h5p-plugin-admin.php [hunk 5]
```diff
@@ -162,6 +165,70 @@ function add_settings_link($links) {
...
+             $journey_user_id
+          ));
+          
+          // Get user data from h5p_contents_user_data table by content ID and user ID
+          $user_data = json_decode($wpdb->get_var($wpdb->prepare(
```
Comment (zachrip) [inline admin/class-h5p-plugin-admin.php:208]: Is `get_var` correct here? As opposed to get_row, etc?

File: admin/class-h5p-plugin-admin.php [hunk 6]
```diff
@@ -162,6 +165,70 @@ function add_settings_link($links) {
...
+    $action = $_GET['action'];
+    $user_data = array();
+
+    if(!isset($action)) {
+      $core->h5pF->setErrorMessage(__('No action set', $this->plugin_slug));
+      return;
+    }
+    
+    if(strcmp($action, 'get_users_data_by_content') !== 0) {
+      $core->h5pF->setErrorMessage(__('Invalid action set', $this->plugin_slug));
```
Comment (zachrip) [inline admin/class-h5p-plugin-admin.php:185]: Is there a reason this error handling is done with this h5p library stuff compared to the catch block using http_response_code?

**Review by wbk — APPROVED** [review]

**Pull request merged**